src/audio-manager.ts

```diff
@@ -0,0 +1,208 @@
+import {
+    _audioContext,
+    audioBuffers,
+} from './audio-listener.js';
+
+export enum SpatializationType {
+    HRFT,
+    EqualPower,
+}
+
+class AudioManager {
+    private preloadedBuffers: {[key: string]: [Promise<AudioBuffer>, number]} = {};
+
+    async load(file: string): Promise<PlayableNode> {
+        try {
+            /* Return when a instance of this file already being loaded */
+            const [bufferPromise, referenceCount] = this.preloadedBuffers[file] || [
+                undefined,
+                0,
+            ];
+            if (bufferPromise !== undefined) {
+                this.preloadedBuffers[file][1] += 1;
+                return bufferPromise.then(() => new PlayableNode(file));
+            }
+            const response = await fetch(file);
+
+            if (!response.ok) {
+                return Promise.reject(`Failed to fetch audio data from ${file}`);
+            }
+
+            /* Create promise that resolves once decoding is complete */
+            const decodingPromise = new Promise<AudioBuffer>((resolve, reject) => {
+                response
+                    .arrayBuffer()
+                    .then((buffer) => _audioContext.decodeAudioData(buffer))
+                    .then((decodedBuffer) => resolve(decodedBuffer))
+                    .catch((error) => reject(error));
+            });
+
+            this.preloadedBuffers[file] = [decodingPromise, 1];
+
+            /* Return a promise that resolves with a PlayableNode when decoding is complete */
+            return decodingPromise.then(() => new PlayableNode(file));
+        } catch (error) {
+            return Promise.reject(`audio-manager: Error in load() for file ${file}`);
+        }
+    }
+
+    // @todo: This function should only be used internally by PlayableNode.
+    async remove(file: string) {
```

Comment:
```suggestion
    /** @hidden */
    private async _remove(file: string) {
```

And I think you can call it via `manager['_remove'](file)`.

Right, @DavidPeicho ?

Reply:
resolved this by doing it similarly as we did in the audio-listener